scripts/pages/pgCustomHeader.ts

```diff
@@ -0,0 +1,68 @@
+import PgCustomHeaderDesign from 'generated/pages/pgCustomHeader';
+import { withDismissAndBackButton } from '@smartface/mixins';
+import { Router, Route } from '@smartface/router';
+import System from '@smartface/native/device/system';
+import Color from '@smartface/native/ui/color';
+import Font from '@smartface/native/ui/font';
+import { FontStyle } from '@smartface/native/ui/font/font';
+
+const isAndroid = System.OS === System.OSType.ANDROID;
+
+export default class PgCustomHeader extends withDismissAndBackButton(PgCustomHeaderDesign) {
+    searchIsActive: boolean = false;
+    constructor(private router?: Router, private route?: Route) {
```

Comment:
It's better to use encapsulation and write the component logic in the component instead of the page.

Also you need to use dispatch&style for the updates.